packages/expo-cli/src/credentials/views/__tests__/SetupAndroidKeystore-test.ts

```diff
@@ -0,0 +1,80 @@
+import { getCtxMock, testExperienceName } from '../../test-fixtures/mocks-android';
+import { SetupAndroidKeystore } from '../SetupAndroidKeystore';
+import { UpdateKeystore } from '../AndroidKeystore';
+
+jest.mock('../../actions/list');
+
+const originalWarn = console.warn;
+const originalLog = console.log;
+beforeAll(() => {
+  console.warn = jest.fn();
+  console.log = jest.fn();
+});
+afterAll(() => {
+  console.warn = originalWarn;
+  console.log = originalLog;
+});
+beforeEach(() => {});
+
+describe('SetupAndroidKeystore', () => {
+  describe('has valid credentials', () => {
+    it('should check credentials and exit', async () => {
+      const ctx = getCtxMock();
+      const view = new SetupAndroidKeystore(testExperienceName, {});
+      const lastView = await view.open(ctx);
+
+      expect(lastView).toBe(null);
+      expect(ctx.android.fetchKeystore.mock.calls.length).toBe(1);
+      expect(ctx.android.updateKeystore.mock.calls.length).toBe(0);
+    });
+  });
+
+  describe('no credentials', () => {
+    it('should check credentials and launch UpdateKeystore view', async () => {
+      const ctx = getCtxMock({
+        android: {
+          fetchKeystore: jest.fn().mockImplementationOnce(() => null),
+        },
+      });
+
+      const view = new SetupAndroidKeystore(testExperienceName, {});
+      const updateView = await view.open(ctx);
+
+      expect(updateView).toBeInstanceOf(UpdateKeystore);
+      expect(ctx.android.fetchKeystore.mock.calls.length).toBe(1);
+    });
+  });
+
+  describe('when non interactive', () => {
```

Comment:
glad we have this test! a minority of users depend on it for their CI jobs, and the `non-interactive` errors take much longer to get reported